### PR TITLE
Change the query timestamps RPC to streaming RPC

### PR DESCRIPTION
**Pull request by RenfengLiu**
Change the RPCs for GAPIT<-->GAPIS and GAPIS<-->GAPIR to streaming
RPC for querying timestamps. This also includes adding a notification
instruction to the VM to streaming the data back.

### CODE REVIEW

File: gapir/README.md [hunk 1]
```diff
@@ -219,7 +219,14 @@ a jump instruction can jump to this instruction and start execution from there.
 `<code:6> <value:26>`
 
 Jump to the instruction specified by the jump label and start execution from there
-if the value on the top of the stack is not zero. Otherwise it is a Nop. 
+if the value on the top of the stack is not zero. Otherwise it is a Nop.
+
+### `NOTIFICATION()` [-2 (uint32_t, pointer)]
+`<code:6> <padding:26>`
+
+Pops size and then a pointer from the top of the stack and stream back size bytes of
```
Comment (pau-baiget) [inline gapir/README.md]: stream -> streams and emphasize `size`

File: gapir/cc/context.cpp [hunk 2]
```diff
@@ -792,4 +797,18 @@ bool Context::stopTimer(Stack* stack, bool pushReturn) {
   return false;
 }
 
+bool Context::sendNotificationData(Stack* stack) {
+  GAPID_WARNING("Sending notification data via gPRC service.");
```
Comment (pau-baiget) [inline gapir/cc/context.cpp]: Should this be sent as a warning?

Comment (AWoloszyn) [reply gapir/cc/context.cpp]: This will spam warnings, which we should avoid.

File: gapir/cc/interpreter.cpp [hunk 3]
```diff
@@ -301,6 +301,11 @@ Interpreter::Result Interpreter::post() {
   return this->call(Interpreter::POST_FUNCTION_ID);
 }
 
+Interpreter::Result Interpreter::notification() {
+  GAPID_WARNING("call notification function");
```
Comment (AWoloszyn) [inline gapir/cc/interpreter.cpp]: Remove warning please.

**Comment by AWoloszyn** [comment]
Other than the couple comments, looks good to me.

**Comment by AWoloszyn** [comment]
LGTM, if you can make sure that "Report" still works. (You will have to set VK_LAYER_PATH=/path/to/validation/layers")

**Comment by RenfengLiu** [comment]
Just tested with the cube by commenting out 
```    
 // vkDestroyShaderModule(demo->device, demo->frag_shader_module, NULL);
 // vkDestroyShaderModule(demo->device, demo->vert_shader_module, NULL);```
 and it can correctly report the issue.

**Comment by AWoloszyn** [comment]
That's perfect. Thanks. Feel free to submit (if you are able to).

**Pull request merged**